File: week01/MergeSortedArray.java
public void merge(int[] nums1, int m, int[] nums2, int n) {
        int left = 0, right = 0, core = 0;
        int[] temp = new int[nums1.length];
        
        for(int i = 0; i<temp.length;i++) temp[i] = nums1[i];
        
        while(left < m && right < n){
            if(temp[left] >= nums2[right]){
                nums1[core++] = nums2[right++];
            }else nums1[core++] = temp[left++];
        }
        
        while(left < m) nums1[core++] = temp[left++];
        while(right < n) nums1[core++] = nums2[right++];
        
    }

File: week01/mergeSortedArray.js
var merge = function (nums1, m, nums2, n) {
  let totalIndex = m + n - 1;
  m = m - 1;
  n = n - 1;

  while (m >= 0 && n >= 0) {
    if (nums1[m] > nums2[n]) {
      nums1[totalIndex] = nums1[m];
      m--;
    } else {
      nums1[totalIndex] = nums2[n];
      n--;
    }
    totalIndex--;
  }

  while (n >= 0) {
    nums1[totalIndex] = nums2[n];
    totalIndex--;
    n--;
  }
};
